yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeSplit.kt

```diff
@@ -0,0 +1,577 @@
+    ): Pair<RgaTreeSplitNode<T>, RgaTreeSplitNode<T>?> {
+        if (candidates.isEmpty()) {
+            error("findEdgesOfCandidates error: candidates is empty")
+        }
+        return Pair(requireNotNull(candidates.first().prev), candidates.last().next)
+    }
+
+    private fun makeChanges(
+        boundaries: List<RgaTreeSplitNode<T>?>,
+        executedAt: TimeTicket,
+    ): List<TextChange> {
+        return buildList {
+            var (fromIndex, toIndex) = Pair(0, 0)
+            for (index in 0..boundaries.size - 2) {
+                val leftBoundary = boundaries[index]
+                val rightBoundary = boundaries[index + 1]
+                if (leftBoundary?.next == rightBoundary) continue
+
+                fromIndex =
+                    findIndexesFromRange(requireNotNull(leftBoundary?.next).createRange()).first
+                toIndex = if (rightBoundary == null) {
+                    treeByIndex.length
+                } else {
+                    findIndexesFromRange(requireNotNull(rightBoundary.prev).createRange()).second
+                }
+            }
+            if (fromIndex < toIndex) {
+                add(TextChange(TextChangeType.Content, executedAt.actorID, fromIndex, toIndex))
+            }
+        }.reversed()
+    }
+
+    fun findIndexesFromRange(range: RgaTreeSplitNodeRange): Pair<Int, Int> {
+        val (fromPos, toPos) = range
+        return Pair(findIndexFromNodePos(fromPos, false), findIndexFromNodePos(toPos, true))
+    }
+
+    private fun findIndexFromNodePos(
+        pos: RgaTreeSplitNodePos,
+        preferToLeft: Boolean,
+    ): Int {
+        val absoluteID = pos.absoluteID
+        val node = if (preferToLeft) {
+            findFloorNodePreferToLeft(absoluteID)
+        } else {
+            findFloorNode(absoluteID)
+        } ?: throw NoSuchElementException("the node of the given ID should be found: $absoluteID")
+
+        val index = treeByIndex.indexOf(node)
+        val offset = if (node.isRemoved) 0 else absoluteID.offset - node.id.offset
+        return index + offset
+    }
+
+    /**
+     * Clears the index nodes of the given deletion [boundaries].
+     * The [boundaries] mean the nodes that will not be deleted in the range.
+     */
+    private fun deleteIndexNodes(boundaries: List<RgaTreeSplitNode<T>?>) {
+        for (index in 0..boundaries.size - 2) {
+            val leftBoundary = boundaries[index]
+            val rightBoundary = boundaries[index + 1]
+            // NOTE(7hong13): below comment has a typo in JS-SDK.
+            // If there is no node to delete between boundaries, do nothing.
+            if (leftBoundary?.next != rightBoundary) {
+                treeByIndex.deleteRange(requireNotNull(leftBoundary), rightBoundary)
+            }
+        }
+    }
+
+    // NOTE(7hong13): original comment from JS-SDK:
+    // `findNodePos` finds RGATreeSplitNodePos of given offset.
+    /**
+     * Finds [RgaTreeSplitNodePos] of the given [index].
```

Comment:
@hackerwins I left some notes on the comments I fixed.
(refer to: [link](https://github.com/yorkie-team/yorkie-js-sdk/blob/893b2c9a870129068d152e092b2908b53fef8e72/src/document/crdt/rga_tree_split.ts#L541))